resolwe_bio/processes/support_processors/prepare_geo.yml

```diff
@@ -0,0 +1,267 @@
+# ===========
+# Prepare GEO
+# ===========
+---
+
+- slug: prepare-geo-chipseq
+  name: Prepare GEO - ChIP-Seq
+  requirements:
+    expression-engine: jinja
+    resources:
+      cores: 1
+    executor:
+      docker:
+        image: resolwebio/base:ubuntu-17.10
+  data_name: 'Prepare GEO ({{name}} - ChIP-Seq)'
+  version: 0.0.1
+  type: data:other:geo:chipseq
+  category: analyses
+  persistence: CACHED
+  description: >
+    Prepare ChIP-Seq data for GEO upload.
+  input:
+    - name: reads
+      label: Reads
+      type: list:data:reads:fastq
+      description: >
+        List of reads objects. Fastq files will be used.
+    - name: macs14
+      label: Macs14
+      type: list:data:chipseq:macs14
+      description: >
+        List of macs14 objects. Wiggle files will be used.
+    - name: relations
+      label: Relations
+      type: list:basic:string
+      description: >
+        Sample to background mapper. Accepted format <sample name>:<background sample name>.
+    - name: name
+      label: Collection name
+      type: basic:string
+  output:
+    - name: tarball
+      label: GEO folder
+      type: basic:file
+    - name: table
+      label: Annotation table
+      type: basic:file
+  run:
+    runtime: polyglot
+    language: bash
+    program: |
+      NAME={{ name }}_ChIP-Seq
+      mkdir "${NAME}"
+
+      {% set macs_first = macs14|first %}
+
+      declare -A SAMPLE_TO_PAIRED
```

Comment:
You can do a pure Jinja implementation of `SAMPLE_TO_PAIRED`, `SAMPLE_TO_BACKGROUND`, and `NAME`.